### PR TITLE
Migrate the framework to Swift. Keep only necessary functions.

### PR DESCRIPTION
**Pull request by lnknguyen**

### CODE REVIEW

File: EYRCall.swift [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                print(error!)
+            }
+        }
+        session = nil
+        
+        hasEnded = true
+    }
+}
+
+extension Call: OTSessionDelegate {
```
Comment (Panda-ref) [inline EYRCall.swift]: Why do we need this delegate?

File: EYRCall.swift [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        
+        hasConnected = false
+        canStartCall?(false)
+        canAnswerCall?(false)
+    }
+    
+    func session(_ session: OTSession, streamCreated stream: OTStream) {
+        print(#function)
+        subscriber = OTSubscriber.init(stream: stream, delegate: self)
+        subscriber?.subscribeToVideo = false
```
Comment (Panda-ref) [inline EYRCall.swift]: What is happening here?

File: README.md [hunk 3]
```diff
@@ -11,18 +11,32 @@ For more information about **ConnectionService** on Android, please see [Android
 
 ⚠️ **CallKit** and **ConnectionService** are only available on real devices, this library will not work on simulators.
 
-# Demo
+#  Summary
```
Comment (Panda-ref) [inline README.md]: Readme is not valid anymore.
Maybe update it?

File: actions.js [hunk 4]
```diff
@@ -1,87 +1,28 @@
...
-const didPerformDTMFAction = handler =>
-  eventEmitter.addListener(RNCallKeepDidPerformDTMFAction, (data) => handler(data));
-
-const didResetProvider = handler =>
-  eventEmitter.addListener(RNCallKeepProviderReset, handler);
+const isIOS = Platform.OS === 'ios';
 
-const checkReachability = handler =>
-  eventEmitter.addListener(RNCallKeepCheckReachability, handler);
+const answerCall = handler => eventEmitter.addListener(EYRCallKeepPerformAnswerCallAction, (data) => handler(data));
```
Comment (Panda-ref) [inline actions.js]:
```suggestion
const answerCall = handler => eventEmitter.addListener(EYRCallKeepPerformAnswerCallAction, handler);
```

Comment (Panda-ref) [reply actions.js]: same for other places

File: actions.js [hunk 5]
```diff
@@ -1,87 +1,28 @@
...
-const showIncomingCallUi = handler =>
-  eventEmitter.addListener(RNCallKeepShowIncomingCallUi, (data) => handler(data));
+const didPerformSetMutedCallAction = handler => eventEmitter.addListener(EYRCallKeepDidPerformSetMutedCallAction, (data) => handler(data));
 
-const silenceIncomingCall = handler =>
-  eventEmitter.addListener(RNCallKeepOnSilenceIncomingCall, (data) => handler(data));
+const didLoadWithEvents = handler => eventEmitter.addListener(EYRCallKeepDidLoadWithEvents, handler);
 
-export const emit = (eventName, payload) => eventEmitter.emit(eventName, payload);
+export const emit = (eventName, payload) => newEventEmitter.emit(eventName, payload);
```
Comment (Panda-ref) [inline actions.js]: wtf is newEventEmitter

File: index.d.ts [hunk 6]
```diff
@@ -137,6 +142,10 @@ declare module 'react-native-callkeep' {
 
     static getCalls(): Promise<object>
 
+    static getAudioRoutes(): Promise<void>
```
Comment (Panda-ref) [inline index.d.ts]: Let's clean typings up according to the js API we expose

File: index.js [hunk 7]
```diff
@@ -48,78 +50,22 @@ class RNCallKeep {
     return this._setupIOS(options.ios);
   };
 
-  registerPhoneAccount = () => {
```
Comment (Panda-ref) [inline index.js:51]: We don't allow setup from JS anymore. Let's remove it.

File: index.js [hunk 8]
```diff
@@ -48,78 +50,22 @@ class RNCallKeep {
...
-    if (isIOS) {
-      return;
-    }
-    RNCallKeepModule.registerPhoneAccount();
-  };
-
   registerAndroidEvents = () => {
```
Comment (Panda-ref) [inline index.js]: We don't need that atm, let's remove

File: index.js [hunk 9]
```diff
@@ -48,78 +50,22 @@ class RNCallKeep {
...
-      supportsHolding,
-      supportsDTMF,
-      supportsGrouping,
-      supportsUngrouping
-    );
-  };
-
   answerIncomingCall = (uuid) => {
-    RNCallKeepModule.answerIncomingCall(uuid);
+    EYRCallKeepModule.answerIncomingCalld(uuid);
```
Comment (Panda-ref) [inline index.js]: We don't allow answering the call from APP UI.

File: index.js [hunk 10]
```diff
@@ -136,22 +82,17 @@ class RNCallKeep {
...
-
-  reportConnectedOutgoingCallWithUUID = (uuid) => {
-    //only available on iOS
-    if (isIOS) {
-      RNCallKeepModule.reportConnectedOutgoingCallWithUUID(uuid);
-    }
-  };
-
-  reportEndCallWithUUID = (uuid, reason) => RNCallKeepModule.reportEndCallWithUUID(uuid, reason);
+  // Using new module
```
Comment (Panda-ref) [inline index.js]: Such comments don't bring value usually. I suggest removing it.

File: index.js [hunk 11]
```diff
@@ -136,22 +82,17 @@ class RNCallKeep {
     return RNCallKeepModule.isConnectionServiceAvailable();
```
Comment (Panda-ref) [inline index.js]: Remove checkPhoneAccountEnabled, isConnectionServiceAvailable.
Jarvis will work on Android API.

File: index.js [hunk 12]
```diff
@@ -160,7 +101,7 @@ class RNCallKeep {
     if (!isIOS) {
```
Comment (Panda-ref) [inline index.js]: Keep iOS only relevant code

File: index.js [hunk 13]
```diff
@@ -181,17 +121,12 @@ class RNCallKeep {
 
   endAllCalls = () => RNCallKeepModule.endAllCalls();
```
Comment (Panda-ref) [inline index.js]: does not exist at the  bridge level

File: index.js [hunk 14]
```diff
@@ -181,17 +121,12 @@ class RNCallKeep {
...
 
-  supportConnectionService = () => supportConnectionService;
-
-  hasPhoneAccount = async () => (isIOS ? true : await RNCallKeepModule.hasPhoneAccount());
-
   hasOutgoingCall = async () => (isIOS ? null : await RNCallKeepModule.hasOutgoingCall());
```
Comment (Panda-ref) [inline index.js]: does not exist at the bridge level

File: index.js [hunk 15]
```diff
@@ -200,11 +135,9 @@ class RNCallKeep {
...
 
-  checkIfBusy = () =>
-    isIOS ? RNCallKeepModule.checkIfBusy() : Promise.reject('RNCallKeep.checkIfBusy was called from unsupported OS');
+  getAudioRoutes = () => RNCallKeepModule.getAudioRoutes();
 
-  checkSpeaker = () =>
-    isIOS ? RNCallKeepModule.checkSpeaker() : Promise.reject('RNCallKeep.checkSpeaker was called from unsupported OS');
+  setAudioRoute = (uuid, inputName) => RNCallKeepModule.setAudioRoute(uuid, inputName);
 
   setAvailable = (state) => {
```
Comment (Panda-ref) [inline index.js]: Not relevant

Comment (Panda-ref) [reply index.js]: setForegroundServiceSettings not relevant

File: index.js [hunk 16]
```diff
@@ -225,14 +158,6 @@ class RNCallKeep {
...
 
-  canMakeMultipleCalls = (state) => {
-    if (isIOS) {
-      return;
-    }
-
-    RNCallKeepModule.canMakeMultipleCalls(state);
-  };
-
   setCurrentCallActive = (callUUID) => {
```
Comment (Panda-ref) [inline index.js]: not relevant

File: index.js [hunk 17]
```diff
@@ -241,6 +166,34 @@ class RNCallKeep {
     RNCallKeepModule.setCurrentCallActive(callUUID);
   };
 
+    displayIncomingCall = (
```
Comment (Panda-ref) [inline index.js]: We don't allow displaying incoming call from JS

File: index.js [hunk 18]
```diff
@@ -241,6 +166,34 @@ class RNCallKeep {
...
+        EYRCallKeepModule.displayIncomingCall(
+          uuid,
+          handle,
+          handleType,
+          hasVideo,
+          localizedCallerName,
+        );
+      };
+
   updateDisplay = (uuid, displayName, handle, options = null) => {
```
Comment (Panda-ref) [inline index.js]: Not relevant

File: index.js [hunk 19]
```diff
@@ -260,15 +213,6 @@ class RNCallKeep {
```
Comment (Panda-ref) [inline index.js]: not relevant + does not exist on the bridge level

File: index.js [hunk 20]
```diff
@@ -260,15 +213,6 @@ class RNCallKeep {
 
   setReachable = () => RNCallKeepModule.setReachable();
```
Comment (Panda-ref) [inline index.js]: not relevant + does not exist on the bridge level

File: index.js [hunk 21]
```diff
@@ -260,15 +213,6 @@ class RNCallKeep {
...
-  // @deprecated
-  reportUpdatedCall = (uuid, localizedCallerName) => {
-    console.warn('RNCallKeep.reportUpdatedCall is deprecated, use RNCallKeep.updateDisplay instead');
-
-    return isIOS
-      ? RNCallKeepModule.reportUpdatedCall(uuid, localizedCallerName)
-      : Promise.reject('RNCallKeep.reportUpdatedCall was called from unsupported OS');
-  };
-
   _setupIOS = async (options) =>
```
Comment (Panda-ref) [inline index.js]: we don't allow setup from JS

File: index.js [hunk 22]
```diff
@@ -278,7 +222,7 @@ class RNCallKeep {
         reject('RNCallKeep.setup: option "appName" should be of type "string"');
       }
 
-      resolve(RNCallKeepModule.setup(options));
+      resolve(EYRCallKeepModule.setup(options));
     });
 
   _setupAndroid = async (options) => {
```
Comment (Panda-ref) [inline index.js]: not relevant

File: index.js [hunk 23]
```diff
@@ -298,37 +242,7 @@ class RNCallKeep {
...
-            style: 'cancel',
-          },
-          { text: options.okButton, onPress: () => resolve(true) },
-        ],
-        { cancelable: true }
-      );
-    });
-
+    
   backToForeground() {
```
Comment (Panda-ref) [inline index.js]: not relevant

File: ios/RNCallKeep/EYRCallKeep.swift [hunk 24]
```diff
@@ -0,0 +1,355 @@
...
+        // Send event with name wrapper
+        self.sendEventWithNameWrapper(EYRCallKeepDidPerformSetMutedCallAction,
+                                      body: ["callUUID": action.callUUID.uuidString.lowercased()])
+        
+        action.fulfill()
+    }
+    
+    
+    /*
+    public func provider(_ provider: CXProvider, didActivate audioSession: AVAudioSession) {
```
Comment (Panda-ref) [inline ios/RNCallKeep/EYRCallKeep.swift:375]: According to your JS API you allow changing the audioRoute. Shouldn't it be uncommented?

File: ios/RNCallKeep/RNCallKeep.h [hunk 25]
```diff
@@ -6,45 +6,8 @@
...
 
-#import <Foundation/Foundation.h>
-#import <UIKit/UIKit.h>
-#import <CallKit/CallKit.h>
-#import <Intents/Intents.h>
-//#import <AVFoundation/AVAudioSession.h>
-
-#import <React/RCTEventEmitter.h>
-
+/*
```
Comment (Panda-ref) [inline ios/RNCallKeep/RNCallKeep.h:9]: Can't we delete it?

File: ios/RNCallKeep/RNCallKeepBridge.m [hunk 26]
```diff
@@ -0,0 +1,20 @@
...
+//  RNCallKeep
+//
+//  Created by N L on 27.8.2021.
+//  Copyright © 2021 react-native-webrtc. All rights reserved.
+//
+
+#import <React/RCTBridgeModule.h>
+#import <React/RCTEventEmitter.h>
+
+@interface RCT_EXTERN_MODULE(EYRCallKeep, RCTEventEmitter)
```
Comment (Panda-ref) [inline ios/RNCallKeep/RNCallKeepBridge.m:12]: JS API is not in sync with bridge

File: package-lock.json [hunk 27]
```diff
@@ -0,0 +1,5627 @@
+{
```
Comment (Panda-ref) [inline package-lock.json]: Don't lock versions. It's uncommon to react native dependencies

**Review by Panda-ref — CHANGES_REQUESTED** [review]

File: ios/RNCallKeep/RNCallKeepBridge.m [hunk 28]
```diff
@@ -0,0 +1,20 @@
...
+RCT_EXTERN_METHOD(setMutedCall:(NSString *)uuidString muted:(BOOL))
+RCT_EXTERN_METHOD(endCall:(NSString*)uuidString)
+RCT_EXTERN_METHOD(answerIncomingCall:(NSString*)uuidString)
```
Comment (Panda-ref) [inline ios/RNCallKeep/RNCallKeepBridge.m]: We don't allow answering incoming call from APP UI

File: ios/RNCallKeep/RNCallKeepBridge.m [hunk 29]
```diff
@@ -0,0 +1,20 @@
...
+RCT_EXTERN_METHOD(reportEndCall:(NSString*)uuidString reason:(int))
```
Comment (Panda-ref) [inline ios/RNCallKeep/RNCallKeepBridge.m]: inconsistent with the current implementation reportEndCallWithUUID

File: EYRCall.swift (hunk 2 above)
Comment (lnknguyen) [reply EYRCall.swift]: Experimental code committed by mistake, deleted.

File: ios/RNCallKeep/RNCallKeep.h (hunk 25 above)
Comment (lnknguyen) [reply ios/RNCallKeep/RNCallKeep.h:9]: Currently we need to keep this header file to expose the Swift code to ObjC project. I'll look for an alternative solution

File: ios/RNCallKeep/RNCallKeepBridge.m (hunk 29 above)
Comment (lnknguyen) [reply ios/RNCallKeep/RNCallKeepBridge.m:15]: would suggest changing the handler name in index.js: reportEndCallWithUUID -> reportEndCall. see setMutedCall

**Pull request merged**